packages/core/src/use-mutation.ts

```diff
@@ -15,10 +15,13 @@ export function useMutation<TResult, TError = unknown, TVariables = undefined, T
   mutationResolver: MutationFunction<TResult, TVariables>,
   config?: MutationConfig<TResult, TError, TVariables, TSnapshot>,
 ) {
-  validateQueryFn(mutationResolver)
+  const enhancedResolverRpcClient = sanitize(mutationResolver)
 
-  return useReactQueryMutation(mutationResolver, {
-    throwOnError: true,
-    ...config,
-  }) as MutationResultPair<TResult, TError, TVariables, TSnapshot>
+  return useReactQueryMutation(
+    (variables?: TVariables) => enhancedResolverRpcClient(variables, {fromQueryHook: true}),
```

Comment:
Unless I'm mistaking, the previous type for this was `MutationFunction` which is typed `(variables: TVariables)` and so the new code should be like this to match.
```suggestion
    (variables: TVariables) => enhancedResolverRpcClient(variables, {fromQueryHook: true}),
```